solutions/swb-reference/integration-tests/tests/multiStep/project.test.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import ClientSession from '../../support/clientSession';
+import Setup from '../../support/setup';
+
+describe('multiStep project tests', () => {
+  const setup: Setup = new Setup();
+  let adminSession: ClientSession;
+  let costCenterId: string;
+
+  beforeAll(async () => {
+    adminSession = await setup.getDefaultAdminSession();
+    costCenterId = setup.getSettings().get('costCenterId');
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  test('happy path', async () => {
+    console.log('Creating Project');
+    const { data: createdProject } = await adminSession.resources.projects.create({
+      name: 'Dragon Project',
+      description: 'Project for TOP SECRET dragon research',
+      costCenterId
+    });
+
+    console.log('Getting Project');
+    const { data: getProject } = await adminSession.resources.projects.project(createdProject.id).get();
+    expect(getProject).toMatchObject(createdProject);
+
+    console.log('Listing Projects');
+    const { data: listProject } = await adminSession.resources.projects.get({
+      'filter[name][eq]': 'Dragon Project'
+    });
+    expect(listProject.data.length).toEqual(1);
+    expect(listProject.data[0]).toMatchObject(createdProject);
+
+    console.log('Updating Project');
+    const newName = 'Totally Normal Project';
+    const newDescription = 'Not a Project studying dragons!';
+    const { data: updatedProject } = await adminSession.resources.projects
+      .project(createdProject.id)
+      .update({ name: newName, description: newDescription }, true);
+    expect(updatedProject).toMatchObject({ name: newName, description: newDescription });
+
+    console.log('Deleting Project');
+    // eslint-disable-next-line no-unused-expressions
+    expect(await adminSession.resources.projects.project(createdProject.id).softDelete()).resolves;
+
+    console.log("Listing projects doesn't return deleted project");
+    const { data: listProjectAfterDelete } = await adminSession.resources.projects.get({
+      'filter[name][eq]': 'Totally Normal Project'
```

Comment:
could we use `{ filter : { name: { eq: 'Totally Normal Project'}}} `instead

Reply:
that didn't work! I got an error when trying to use that kind of request. This is how the request looks for the API itself in the query params.